goal_src/levels/beach/lurkercrab.gc

```diff
@@ -5,3 +5,1438 @@
+     :flags #x1
+     :random-range #x47800000
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec
+     :field #x3e
+     :flags #x1
+     :initial-value #x444ccccd
+     :random-mult #x3f800000
+     )
+    (new 'static 'sp-field-init-spec :field #x43)
+    )
+   )
+  )
+
+;; definition of type lurkercrab
+(deftype lurkercrab (nav-enemy)
+  ((orient basic  :offset-assert 400)
+   )
+  :heap-base #x130
+  :method-count-assert 76
+  :size-assert         #x194
+  :flag-assert         #x4c01300194
+  )
+
+;; failed to figure out what this is:
+(let
+  ((v1-4
+    (new 'static 'skeleton-group
+     :art-group-name "lurkercrab"
+     :bounds (new 'static 'vector :w 10240.0)
+     :max-lod 2
+     :version #x6
+     )
+    )
+   )
+  (set! (-> v1-4 jgeo) 0)
+  (set! (-> v1-4 janim) 4)
+  (set! (-> v1-4 mgeo 0) (the-as uint 1))
+  (set! (-> v1-4 lod-dist 0) 81920.0)
+  (set! (-> v1-4 mgeo 1) (the-as uint 2))
+  (set! (-> v1-4 lod-dist 1) 163840.0)
+  (set! (-> v1-4 mgeo 2) (the-as uint 3))
+  (set! (-> v1-4 lod-dist 2) 4095996000.0)
+  (set! *lurkercrab-sg* v1-4)
+  )
+
+;; definition for method 44 of type lurkercrab
+;; INFO: Return type mismatch none vs object.
+(defmethod
+  dummy-44
+  lurkercrab
+  ((obj lurkercrab) (arg0 process) (arg1 event-message-block))
+  (if
+   (and
+    (logtest? (-> obj nav-enemy-flags) 64)
+    ((method-of-type touching-shapes-entry prims-touching?)
+     (the-as touching-shapes-entry (-> arg1 param 0))
+     (-> obj collide-info)
+     (the-as uint 1)
+     )
+    )
+   (nav-enemy-send-attack
+    arg0
+    (the-as touching-shapes-entry (-> arg1 param 0))
+    'generic
+    )
+   )
+  (let ((t9-2 (method-of-object (-> obj collide-info) dummy-55)))
+   (-> arg1 param 0)
+   0.7
+   6144.0
+   16384.0
+   (t9-2)
+   )
+  (the-as object (if (zero? (logand (-> obj nav-enemy-flags) 256))
+                  (dummy-45 (-> obj collide-info))
+                  )
+   )
+  )
+
+;; definition for method 43 of type lurkercrab
+;; INFO: Return type mismatch none vs object.
+;; WARN: rewrite_to_get_var got a none typed variable. Is there unreachable code?
+;; WARN: rewrite_to_get_var got a none typed variable. Is there unreachable code?
+;; WARN: rewrite_to_get_var got a none typed variable. Is there unreachable code?
+;; WARN: rewrite_to_get_var got a none typed variable. Is there unreachable code?
+;; Used lq/sq
+(defmethod
+  dummy-43
+  lurkercrab
+  ((obj lurkercrab) (arg0 process) (arg1 event-message-block))
+  (let ((s5-0 (-> obj incomming-attack-id)))
+   (set! (-> obj incomming-attack-id) (the-as handle (-> arg1 param 2)))
+   (let ((v1-1 (-> arg1 param 1)))
+    (the-as object (cond
+                    ((or (= v1-1 'flop) (= v1-1 'explode) (= v1-1 'darkeco))
+                     (logclear! (-> obj mask) (process-mask actor-pause))
+                     (go (method-of-object obj nav-enemy-die))
+                     )
+                    ((= v1-1 'punch)
+                     (the-as none (cond
```

Comment:
hm this `(the-as none` isn't great.  I may need to work on the decompiler a bit to make this work properly.  But this shouldn't block this PR from merging.